File: backend/semgrep_rules/tests/java/kisa_sw_38_improper_exception_handling.java
class kisa_sw_38_improper_exception_handling {
    Object unsafe() {
        try {
            return risky();
        } catch (Exception e) {
            // ruleid: kisa.sw38.java.overbroad-exception-default-return
            return null;
        }
    }
    Object safe() {
        // ok: kisa.sw38.java.overbroad-exception-default-return
        return risky();
    }
    Object risky() { return new Object(); }
}

File: backend/semgrep_rules/tests/javascript/kisa_sw_38_improper_exception_handling.js
function unsafe() {
  try {
    return risky();
  } catch (e) {
    // ruleid: kisa.sw38.javascript.overbroad-exception-default-return
    return null;
  }
}
function safe() {
  // ok: kisa.sw38.javascript.overbroad-exception-default-return
  return risky();
}
function risky() { return {}; }
